RecoEgamma/EgammaElectronProducers/plugins/TrackingRegionsFromSuperClustersProducer.h

```diff
@@ -247,15 +250,11 @@ getVtxPos(const edm::Event& iEvent,double& deltaZVertex)const
 {
   if(useZInVertex_){
     auto verticesHandle = getHandle(iEvent,verticesToken_);   
-    if(!verticesHandle->empty()){
-      deltaZVertex = originHalfLength_;
-      const auto& pv = verticesHandle->front();
-      return GlobalPoint(pv.x(),pv.y(),pv.z());
-    }
+    deltaZVertex = originHalfLength_;
+    const auto& pv = verticesHandle->front();
+    return GlobalPoint(pv.x(),pv.y(),pv.z());
```

Comment:
so I have thought about how I want to do this. I dislike the beamspot fall back as now we required the beamspot to be there "sometimes".  So I'm going to fall back to  0,0,defaultZ_